Doc install

antoine-de opened the pull request:
add an install script that:
- build navitia
- import data (by default it download a Paris dataset)
- run a front end and a kraken core

it is made more as an installation guide, but can be used as an out of the box build script

Note: I don't understand why the kraken is not killed at the end of the run.
Anybody has an idea ?

antoine-de committed:
Doc: Install script

antoine-de committed:
Doc: test on vm

antoine-de committed:
Doc: handle params

antoine-de committed:
Doc: test on vm

still need to do the kraken/jormun part

antoine-de committed:
Doc: custom install build

antoine-de commented:
Jenkins, test this please

kinnou02 commented on build_navitia.sh:
```diff
@@ -0,0 +1,255 @@
+#Building
+#========
+#
+#First you need to install all dependecies. 
```
typo

kinnou02 commented on build_navitia.sh:
```diff
@@ -0,0 +1,255 @@
+
+kraken_db_name='navitia'
+# for the default build we give ownership of the base to a 'navitia' user, but you can do whatever you want here
+sudo su postgres -c "psql postgres -tAc \"SELECT 1 FROM pg_roles WHERE rolname='$db_owner'\""  # we check if there is already a user 
```
i don't really like the sudo in a script, because I always need to read it before entering my password.
yes, I am a little paranoid

antoine-de replied:
I do agree, but then how can you manage the postgres stuff ?

kinnou02 commented on build_navitia.sh:
```diff
@@ -0,0 +1,255 @@
+sudo su postgres -c "psql postgres -tAc \"SELECT 1 FROM pg_roles WHERE rolname='$db_owner'\""  # we check if there is already a user 
+if [ ! $? ]
+then
+sudo su postgres -c "psql -c \"create user $db_owner;alter user $db_owner password '$kraken_db_user_password';\""
```
the `createuser` shell command would probably be easier. there is a `createdb` too

antoine-de replied:
you cant give easily a password with createuser, so the psql  stuff is easier.

Ok for the createdb

kinnou02 commented on build_navitia.sh:
```diff
@@ -0,0 +1,255 @@
+[LOG]
+log4cplus.rootLogger=DEBUG, ALL_MSGS
+
+log4cplus.appender.ALL_MSGS=log4cplus::FileAppender
```
for testing purpose it can be easier to log on stdout

antoine-de replied:
sure, but since Jormun and kraken are lanched in the same shell the output can become quite messy so I thought It would be better to output Jormungandr and kraken in files

kinnou02 commented on build_navitia.sh:
```diff
@@ -0,0 +1,255 @@
+    # then you need to install all python dependencies: ::
+    
+    sudo pip install -r $navitia_dir/source/jormungandr/requirements.txt
+    sudo pip install -r $navitia_dir/source/tyr/requirements.txt
```
Tyr is not used

antoine-de committed:
Doc: corrections after review

antoine-de committed:
Doc: correction after review

the kraken is correctly killed but I could not make the trap stuff work
so I kill jormungandr and kraken manually

antoine-de committed:
Doc: small lint
